### ISSUE TITLE
received nan in training MAC Transformer

### ISSUE THREAD

**Issue by nargesmokhtari** [lucidrains/titans-pytorch#53]
When training `MemoryAsContextTransformer` from titans-pytorch on enwik8, the training loss becomes `NaN` after a few iterations.  
This happens even at step 0–10 and prevents any meaningful training.
